### PR TITLE
fix(updater): cross-platform compatibility, robustness enhancements

### PR DESCRIPTION
**Pull request by mistydemeo**
This makes a few changes to the updater with the goal of ensuring it runs across all platforms and that it requires fewer external tools. Since these tools aren't preinstalled on Windows, it should ensure that running on Windows just works the same as on other platforms. In particular:

* The external `curl` call has been replaced by a call to `reqwest`, which was already used in the crate for other things. This should also ensure it works on minimal Linux docker images, some of which come with `wget` but no `curl` preinstalled.
* The calls to Unix commands to move and rename files have been replaced with calls to Tokio's `async_fs`.
* Executable file permissions are now handled via Rust stdlib calls, and are only run on Unix; these permissions don't exist on Windows.
* The temporary paths to which tarballs are downloaded are now deleted after being unpacked.

## Summary by CodeRabbit

- **New Features**
    - Enhanced the updater tool to automatically clean up temporary files after use.
    - Improved error handling during file downloads and operations.
- **Refactor**
    - Updated file downloading, moving, and permissions setting methods for better performance and reliability.

### CODE REVIEW

File: crates/cli/src/updater.rs [hunk 1]
```diff
@@ -444,17 +447,14 @@ impl Updater {
...
-            .arg(&artifact_url)
-            .output()
-            .await?;
-
-        if !cmd.status.success() {
-            bail!("Failed to download artifact: {:?}", cmd);
+        match reqwest::get(&artifact_url).await {
+            Ok(response) => {
+                let contents = response.bytes().await?.to_vec();
+                async_fs::write(&target_path, contents).await?;
```
Comment (morgante) [inline crates/cli/src/updater.rs:453]: Is there a way to stream directly into the file, so we don't have to load it all into memory?

**Review by morgante — APPROVED** [review]

File: crates/cli/src/updater.rs (hunk 1 above)
Comment (mistydemeo) [reply crates/cli/src/updater.rs:453]: We can do that! Streaming support is an extra feature in reqwest; are you okay with this expanding the dependency tree a little?

Comment (morgante) [reply crates/cli/src/updater.rs:453]: That's fine, we're not very disciplined about binary size.

Comment (mistydemeo) [reply crates/cli/src/updater.rs]: All right, pushed a change that streams the response and writes chunks to disk.

Comment (morgante) [reply crates/cli/src/updater.rs]: Thanks, looks good!

**Review by morgante — APPROVED** [review]

**Pull request merged**